treeherder/log_parser/failureline.py

```diff
@@ -94,13 +96,22 @@ def get_kwargs(failure_line):
             if key in failure_line}
 
 
+def create_failure_line(job_log, failure_line):
+    fl = FailureLine.objects.create(repository=job_log.job.repository,
+                                    job_guid=job_log.job.guid,
+                                    job_log=job_log,
+                                    **get_kwargs(failure_line))
+    if "group" in failure_line:
+        # Store as unix path separators and omit the filename
+        group, saved = Group.objects.get_or_create(
+            name=path.dirname(path.join(*failure_line["group"].split("\\"))))
```

Comment:
Talked to @edmorley and @ahal on Vidyo a few minutes ago.  Decided we will check the format of the ``group``.  If it's got any problems, we'll still ingest it, but log to new relic that we hit an ugly one.

@ahal says that the filename is generated by him, not the test author, so any new tests will get this proper formatting (forward slashes, relative path).
@ahal has also agreed to back-port this change to ESR.

@ahal and I talked about the field length a bit.  He'd prefer to log the entire field length, even if it exceeds the max that treeherder can store.  This is because the test user looking at the log should be able to get the whole thing.

For Treeherder, this table won't be that large, so I'll go ahead and store as much of the path as I can and truncate if it's too long.  Setting the field length to 255 shouldn't be that big a hit on table size since it'll likely only be a few thousand rows.